rdrf/rdrf/admin.py

```diff
@@ -130,104 +174,142 @@ def export_registry(registry, request):
-        response['Content-Disposition'] = 'attachment; filename="%s"' % name
+            zf.close()
+            zipped_file.flush()
+            zipped_file.seek(0)
 
-        return response
+            response = HttpResponse(FileWrapper(zipped_file), content_type='application/zip')
+            name = "export_" + export_time + "_" + \
+                   reduce(lambda x, y: x + '_and_' + y, [r.code for r in registries]) + ".zip"
+            response['Content-Disposition'] = 'attachment; filename="%s"' % name
 
+            return response
+    export_registry_action.short_description = _("Export")
 
-export_registry_action.short_description = "Export"
+    def generate_questionnaire_action(self, request, registry_models_selected):
+        for registry in registry_models_selected:
+            registry.generate_questionnaire()
 
+    generate_questionnaire_action.short_description = _("Generate Questionnaire")
 
-def design_registry_action(modeladmin, request, registry_models_selected):
-    if len(registry_models_selected) != 1:
-        return
-    else:
-        registry = [r for r in registry_models_selected][0]
-        return HttpResponseRedirect(reverse('rdrf_designer', args=(registry.pk,)))
+    def enable_registration_action(self, request, registry_models_selected):
+        unchanged = []
 
+        for registry in registry_models_selected:
+            if registry.has_feature(RegistryFeatures.REGISTRATION):
+                unchanged.append(registry.code)
 
-design_registry_action.short_description = _("Design")
+            registry.add_feature(RegistryFeatures.REGISTRATION)
+            registry.save()
 
+            messages.success(request, f"Registration enabled for '{registry.code}'")
 
-def generate_questionnaire_action(modeladmin, request, registry_models_selected):
-    for registry in registry_models_selected:
-        registry.generate_questionnaire()
+            existing_notifications = EmailNotification.objects.filter(
+                registry=registry,
+                description__in=EventType.REGISTRATION_TYPES
+            )
 
+            if len(existing_notifications) == 0:
+                messages.warning(request, f"Notifications need to be created for registry '{registry.code}'. "
+                                          f"Use the 'Create notifications' action")
 
-generate_questionnaire_action.short_description = _("Generate Questionnaire")
+        # Check for missing cache table
+        for cache_alias in settings.CACHES:
+            cache = caches[cache_alias]
+            if hasattr(cache, '_table') and cache._table not in connection.introspection.table_names():
+                messages.error(request, _(f"Missing cache table '{cache._table}'. Run django-admin createcachetable"))
 
+        if len(unchanged) > 0:
+            messages.info(request, f"'{', '.join(code for code in unchanged)}' already enabled registration")
```

Comment:
If registration is already enabled the UI currently displays both the "Registration enabled for 'REG_CODE'" and the "REG_CODE" already enabled registration".
In the case, we haven't touched anything we should say just that.